lib/logger/logger.go

```diff
@@ -34,8 +33,24 @@ const (
 const (
 	DefaultFlags = log.Ltime | log.Ldate
 	DebugFlags   = log.Ltime | log.Ldate | log.Lmicroseconds | log.Lshortfile
+	LevelError   = NumLevels // Not really a log level, as it disables all logging.
```

Comment:
I'd prefer to have something dedicated to disabling logging, not using the fact that we don't do error logging as a standin. I was initially confused, as I did expect error logging to exist. And obviously not your PR, but I think it should exist: Imo it would make sense to use an error level for fatal/grave errors distinct from unexpected, but recoverable/handleable errors - currently that's all warning level. And using error level for disabling in this PR likely makes that harder, when/if we want to use error level logging in the future. I haven't checked the diff properly, but from a brief look it doesn't seem like you need to change much, if any logic, it might just be a comment and possibly naming issue to avoid equating error logging to disabling logging entirely.

Reply:
@imsodin I agree 100%. I will change it to `LevelDisabled`. I didn't want to reuse `NumLevels` as that doesn't describe its function in this context.

With this PR's changes, it would be trivial to add an actual `LevelError` or `LevelFatal`. I didn't want to expand the scope of this PR, but happy to do so.

Reply:
Very much agree with keeping the scope, not mixing in more things.